src/main/java/com/icthh/xm/uaa/security/DomainTokenServices.java

```diff
@@ -220,6 +227,26 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
         return accessToken;
     }
 
+    /**
+     * Ensures that {@code user} by {@link Authentication#getPrincipal()} is activated
+     *
+     * @param authentication the authentication token
+     * @throws InvalidTokenException            in case the authentication token is invalid
+     * @throws UserDeniedAuthorizationException in case {@code user} not exist or not activated
+     * @see User#isActivated()
+     */
+    private void verifyUserIsActivated(Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        if (!(principal instanceof String)) {
+            throw new InvalidTokenException("Invalid principal");
+        }
+        String userLogin = (String) principal;
+        boolean isActivated = userService.findOneByLogin(userLogin).map(User::isActivated).orElse(false);
+        if (!isActivated) {
+            throw new UserDeniedAuthorizationException("User is deactivated");
```

Comment:
please extends, and user specific exception for this case

Reply:
Sure, It's better to use `UserNotActivatedException`. Done.